harmony_netcdf_to_zarr/download_utilities.py

```diff
@@ -55,14 +55,21 @@ def download_granules(netcdf_urls: List[str], destination_directory: str,
             download_process.start()
 
         # Ensure worker processes exit successfully
+        exit_codes = []
         for download_process in processes:
             download_process.join()
+            exit_codes.append(download_process.exitcode)
```

Comment:
This is a nice way to grab those pesky `-9` exit codes that weren't being handled before!

I wonder if something relating to the exit code could/should be added to the `shared_namespace`, when it's non-zero? If you have 500 granules and all but one of the processes exit with a `-9` on their first granules, you'll have that one poor old `Process` continuing to churn through the remaining 496 granules, just for the request to presumably fail much later (if I'm understanding this correctly). Adding something to the namespace somehow (like the exception message for native Python `Exception`s) would allow for the other processes to know to stop early.

Reply:
These comments are applicable to both of the instances of `multiprocessing` in the repo (the download worker and output worker stuff)